src/psij/executors/batch/slurm/slurm.mustache

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+
+{{#job.name}}
+#SBATCH --job-name="{{.}}"
+{{/job.name}}
+
+{{#job.spec.directory}}
+#SBATCH --chdir="{{.}}"
+{{/job.spec.directory}}
+
+{{#job.spec.inherit_environment}}
+#SBATCH --export=ALL
+{{/job.spec.inherit_environment}}
+{{^job.spec.inherit_environment}}
+#SBATCH --export=NONE
+{{/job.spec.inherit_environment}}
+
+{{#job.spec.resources}}
+    {{#exclusive_node_use}}
+#SBATCH --exclusive
+    {{/exclusive_node_use}}
+
+    {{#node_count}}
+#SBATCH --nodes={{.}}
+    {{/node_count}}
+
+    {{#process_count}}
+#SBATCH --ntasks={{.}}
+    {{/process_count}}
+
+    {{#processes_per_node}}
+#SBATCH --ntasks-per-node={{.}}
+    {{/processes_per_node}}
+
+    {{#gpu_cores_per_process}}
+#SBATCH --gpus_per_task={{.}}
+    {{/gpu_cores_per_process}}
+
+    {{#cpu_cores_per_process}}
+        {{!SLURM has different meanings for 'CPU', depending on the exact allocation plugin being
+        used. The default plugin uses the literal meaning for 'CPU' - physical processor. The
+        consumable resource allocation plugin, on the other hand, equates 'CPU' to a thread (if
+        hyperthreaded CPUs are used) or CPU core (for non-hyperthreaded CPUs).}}
+#SBATCH --cpus_per_task={{.}}
+    {{/cpu_cores_per_process}}
+{{/job.spec.resources}}
+
+{{#job.spec.attributes}}
+    {{#duration}}
+#SLURM --time={{#psij.lib.walltime_to_minutes}}{{duration}}{{/psij.lib.walltime_to_minutes}}
+    {{/duration}}
+
+    {{#queue_name}}
+#SLURM --partition="{{.}}"
+    {{/queue_name}}
+
+    {{#project_name}}
+#SLURM --account="{{.}}"
+    {{/project_name}}
+
+    {{#reservation_id}}
+#SLURM --reservation="{{.}}"
+    {{/reservation_id}}
+
+    {{#custom_attributes.slurm}}
+#SBATCH --{{key}}="{{value}}"
+    {{/custom_attributes.slurm}}
+
+{{/job.spec.attributes}}
+
+{{!redirect output here instead of through #SBATCH directive since SLURM_JOB_ID is not available
+when the directives are evaluated; the reason for using the job id in the first place being the
+same as for the exit code file.}}
+exec &>> "{{psij.script_dir}}/$SLURM_JOB_ID.out"
+
+{{#psij.launch_command}}{{.}}{{/psij.launch_command}}
+
+{{!we redirect to a file tied to the native ID so that we can reach the file with attach().}}
```

Comment:
That makes sense - but per an earlier comment, the mapping to the native job ID can conflict for multiple re-attached jobs anyway.  

I am not exactly sure how to resolve this, and my suggestions in this review may not be consistent in that respect - apologies for that!   I guess I am just saying that we should make sure that this (pseudo)code is covered:

```py
jex = jpsi.Executor(...)
j1 = jex.submit()
j2 = jex.attach(j1.native_id)
j3 = jex.attach(j2.native_id)
j3.wait()
print(j2.stdout)
print(j3.stdout)
```

I am not sure it is.

Reply:
This conflict would be fine if aux files weren't deleted after being read.
Maybe some form of reference counting would help here.

Reply:
Should work now since aux files are read/deleted only once per native_id.